**Association between Device Type and Customer Satisfaction**

**Background:**

Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights.
They want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.

**Data Provided:**

The data is summarized in a contingency table showing the counts of customers in each satisfaction level for both types of devices:

Satisfaction | Smart Thermostat | Smart Light |	Total
------------ | ---------------- | ----------- | -----
Very Satisfied |	50 |	70 |	120
Satisfied |	80 |	100 |	180
Neutral |	60 |	90 |	150
Unsatisfied |	30 |	50 |	80
Very Unsatisfied |	20 |	50 |	70
Total |	240 |	360 |	600

## Objective:

To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level

## Steps to perform test

## Association between Device Type and Customer Satisfaction
## H0=if there is not a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.
## Ha=if theres a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

In [5]:
import numpy as np
from scipy.stats import chi2_contingency

## Compute Chisquare statistics
alpha = 0.05
CI = 95%
## formulaa used
χ2=∑(𝑂𝑖−𝐸𝑖)2 / 𝐸𝑖


where Oi is the observed frequency and Ei is the expected frequency


## determine the DF
**Determine the degrees of freedom (df)**

$df = (r -1) * (c-1)$

where r is the number of rows and c is the number of columns.

## Compare chisquare statistics:to the critical value from chisquare distribution table at choosen significance level(usually=0.05)

##Lets perform test

**Step1: Calculate Expected Frequencies**
    
The Expected Frequencies calculated for each cell as:
    
$E_{ij} = \frac{(Row Total) * (Column Total)}{Grand Total}$
    

## Step 2: compute chisquare statistics

In [15]:
# Observed frequencies
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

# Perform the Chi-Square test
chi2, p, dof, expected = chi2_contingency(observed)

print("Chi-Square statistic (χ²):", chi2)
print("p-value:", p)
print("Degrees of freedom (df):", dof)
print("Expected frequencies:")
print(expected)

Chi-Square statistic (χ²): 5.638227513227513
p-value: 0.22784371130697179
Degrees of freedom (df): 4
Expected frequencies:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


## Interpretation
p-value is 0.2278 & significance value is 0.05

p-value > significance level

means fail to reject H0 hypothesis

## Conclusion:

There is no statistically significant association between the type of smart home device (Smart Thermostats vs. Smart Lights) and the customer satisfaction level. The distribution of satisfaction levels appears to be independent of the type of device purchased